core/src/main/java/org/elasticsearch/common/geo/builders/CoordinateCollection.java

```diff
@@ -21,24 +21,31 @@
 
 import com.vividsolutions.jts.geom.Coordinate;
 
+import org.elasticsearch.common.io.stream.NamedWriteable;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * The {@link CoordinateCollection} is an abstract base implementation for {@link LineStringBuilder} and {@link MultiPointBuilder}.
  * It holds a common list of {@link Coordinate}, provides setters for adding elements to the list and can render this to XContent.
  */
-public abstract class CoordinateCollection<E extends CoordinateCollection<E>> extends ShapeBuilder {
+public abstract class CoordinateCollection<E extends CoordinateCollection<E>> implements NamedWriteable, ToXContentObject {
```

Comment:
What is the reason this is moving up in the class hierarchy? I see that this used to be a supertype of LineStringBuilder and MultiPointBuilder, maybe to group common functionality. Does CoordinateCollection provide that functionality to all subclasses now? Can we maybe merge ShapeBuilder and CoordinateCollection now? I see that all ShapeBuilders are now also CoordinateCollections, maybe we should only keep ShapeBuilders?

Reply:
Yeah, merging would probably be the best idea to eliminate superfluous classes. I'll have a look at making that change.